### PR TITLE
[Merged by Bors] - feat(Order/Max): add `IsTop.isMax_iff` and `IsBot.isMin_iff`

### PR DESCRIPTION
**Pull request by pitmonticone**
Upstreamed from the [Carleson](https://github.com/fpvandoorn/carleson) project. 

Co-authored-by: Floris van Doorn

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Max.lean [hunk 1]
```diff
@@ -200,6 +200,10 @@ protected theorem IsBot.isMin (h : IsBot a) : IsMin a := fun b _ => h b
 
 protected theorem IsTop.isMax (h : IsTop a) : IsMax a := fun b _ => h b
 
+theorem IsTop.isMax_iff {α} [PartialOrder α] {i j : α} (h : IsTop i) : IsMax j ↔ j = i := by
```
Comment (jcommelin) [inline Mathlib/Order/Max.lean:203]: Could you please add the dual statement as well, for `IsBot` and `IsMin`?

Comment (pitmonticone) [reply Mathlib/Order/Max.lean:203]: Done, thanks!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Pull request closed without merging**